conan/cli/commands/remove.py

```diff
@@ -65,7 +67,7 @@ def remove(conan_api: ConanAPI, parser, *args):
     cache_name = "Local Cache" if not remote else remote.name
 
     def confirmation(message):
-        return args.confirm or ui.request_boolean(message)
+        return not args.dry_run and (args.confirm or ui.request_boolean(message))
```

Comment:
I am not sure the ``dry_run`` should skip the confirmation, maybe it should be done after the ``ui.request``?